platform/ios/src/MGLMapboxEvents.m

```diff
@@ -485,10 +500,15 @@ - (void)postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             if (error) {
                 [strongSelf pushDebugEvent:MGLEventTypeLocalDebug withAttributes:@{MGLEventKeyLocalDebugDescription: @"Network error",
                                                                                         @"error": error}];
+                [[UIApplication sharedApplication] endBackgroundTask:_backgroundTaskIdentifier];
+                _backgroundTaskIdentifier = UIBackgroundTaskInvalid;
                 return;
```

Comment:
If we removed this `return` and then enclosed the subsequent `pushDebugEvent` in an `else` statement, I think we could avoid duplicating the background task invalidation.